vnext/local-cli/config/dependencyConfig.js

```diff
@@ -1,102 +1,244 @@
-  const buf = fs.readFileSync(packageIDL, 'utf8');
-  const indexofNameSpace = buf.indexOf('namespace') + 9;
-  const indexofBracket = buf.indexOf('{');
-  const packageName = buf.substring(indexofNameSpace, indexofBracket).replace(/\s+/g, ' ').trim();
+      const projectFile = path.join(sourceDir, project.projectFile);
 
-  return packageName;
-}
+      const projectContents = configUtils.readProjectFile(projectFile);
 
-// read visual studio project file which is actually a XML doc
-function readProject(projectPath) {
-  return new (xmldoc.XmlDocument)(fs.readFileSync(projectPath, 'utf8'));
-}
+      // Calculating (auto) items
+      project.projectName = configUtils.getProjectName(projectContents);
+      project.projectLang = configUtils.getProjectLanguage(projectFile);
+      project.projectGuid = configUtils.getProjectGuid(projectContents);
 
-function findCppProject(folder, projectName) {
-  const cppProj = glob.sync(path.join('**', projectName + '.vcxproj'), {
-    cwd: folder,
-    ignore: ['node_modules/**', '**/Debug/**', '**/Release/**', '**/Generated Files/**', '**/packages/**'],
-  })[0];
+      if (project.directDependency) {
+        // Calculating more (auto) items
 
-  return cppProj ? path.join(folder, cppProj) : null;
-}
+        const projectNamespace = configUtils.getProjectNamespace(
+          projectContents,
+        );
+        const cppNamespace = projectNamespace.replace(/\./g, '::');
+        const csNamespace = projectNamespace.replace(/::/g, '.');
+
+        project.cppHeaders = project.cppHeaders || [`winrt/${csNamespace}.h`];
+        project.cppPackageProviders = project.cppPackageProviders || [
+          `${cppNamespace}::ReactPackageProvider`,
+        ];
+        project.csNamespaces = project.csNamespaces || [`${csNamespace}`];
+        project.csPackageProviders = project.csPackageProviders || [
+          `${csNamespace}.ReactPackageProvider`,
+        ];
+      }
+    }
+  } else {
+    // No react-native.config, try to heurestically find any projects
+
+    const foundProjects = configUtils.findDependencyProjectFiles(sourceDir);
+
+    for (const foundProject of foundProjects) {
+      const projectFile = path.join(sourceDir, foundProject);
 
-function findCSProject(folder, projectName) {
-  const cppProj = glob.sync(path.join('**', projectName + '.csproj'), {
-    cwd: folder,
-    ignore: ['node_modules/**', '**/Debug/**', '**/Release/**', '**/Generated Files/**', '**/packages/**'],
-  })[0];
+      const projectLang = configUtils.getProjectLanguage(projectFile);
+
+      const projectContents = configUtils.readProjectFile(projectFile);
+
+      const projectName = configUtils.getProjectName(projectContents);
+
+      const projectGuid = configUtils.getProjectGuid(projectContents);
+
+      const projectNamespace = configUtils.getProjectNamespace(projectContents);
+
+      const directDependency = true;
+
+      const cppNamespace = projectNamespace.replace(/\./g, '::');
```

Comment:
I _think_ @dannyvv  made it so the project namespace always uses dots so this could be simplified if you wanted

Reply:
I just wanted to cover my bases.